packages/components/src/button/style.scss

```diff
@@ -237,6 +234,11 @@
 		&:focus {
 			border-radius: $radius-block-ui;
 		}
+
+		&:disabled,
+		&[aria-disabled="true"] {
+			color: $gray-600;
```

Comment:
If we wanted to retain the original design, we could also use [`color-mix()](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix)` to add an exact amount of opacity to the default text color

Reply:
Also, looking at other parts in the repo, `$gray-700` could be a better value here, should we not go for the `color-mix()` and opacity route.

Reply:
It's not entirely clear to me why the default and `secondary` variant use inconsistent color methods (opacity vs specific value). The opacity method seems to pre-date the specific value, so it's probably fine to change this up.

Reply:
`$gray-700` would raise the color contrast to be more visible than I think it should be for a disabled control; we still need to pretty clearly convey that this control is disabled.

Reply:
If I understand correctly, then, you suggest using solid color? Which one would be the correct one, the 600 or 700 gray variant? (or any other ?)

Reply:
One of the reasons why I would prefer solid colors over colors with opacity or alpha transparency is that measuring the color contrast ratio is way easier with solid colors. @jameskoster I understand some blocks e.g. the cover text need opacity/alpha but for user interface controls colors I'd prefer to always use solid colors. Any reason, from a design perspective, to not do that?

Reply:
A solid color seems more intentional, but might not work as well when the button appears on a non-white background. However opacity isn't perfect in that regard either, so changing this is probably okay. gray-600 seems a good fit.

Reply:
As we start adopting more structured theming systems (not necessarily the [Theme component](https://wordpress.github.io/gutenberg/?path=/docs/components-theme--docs), but any system that would allow arbitrary background colors), it would definitely be easier to programatically ensure sufficient contrast if we used solid colors. So if contrast is ever a concern, I would recommend avoiding `opacity`.

Reply:
Thanks @mirka, that's a very good point I'd support it totally.